### ISSUE TITLE
Detect and report mutually-recursive type aliases

### ISSUE THREAD

**Issue by evancz** [elm/compiler#625]
As @mgold points out in issue #623, you get a stack overflow on mutually-recursive type aliases:

``` haskell
type Edge = {vs : (Vertex, Vertex)}
type Vertex = {es : [Edge]}
```

We should detect this and suggest using an ADT, just like in the case of self-recursive type aliases.
